tests/filecheck/dialects/scf/canonicalize.mlir

```diff
@@ -114,3 +114,55 @@ func.func @execute_region_with_multiple_blocks() -> i32 {
   }
   func.return %d : i32
 }
+
+// CHECK-LABEL: func.func @test_true
+// CHECK: %{{.*}} = arith.constant 1 : i32
```

Comment:
I have a feeling this check will pass even if the pattern does nothing. Can you please check whether this is the case? Probably adding a CHECK-NOT for the if operation should do it.